FinSearchUnified/Bundles/SearchBundle/Condition/HasActiveChildCategoryOfCurrentShopCondition.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace FinSearchUnified\Bundles\SearchBundle\Condition;
+
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+
+class HasActiveChildCategoryOfCurrentShopCondition implements ConditionInterface, \JsonSerializable
+{
+    private $shopCategoryId;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'has_active_child_category_of_current_shop_condition';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function jsonSerialize()
+    {
+        return get_object_vars($this);
+    }
+
+    /**
+     * HasActiveCategoryCondition constructor.
+     *
+     * @param int $shopCategoryId
+     */
+    public function __construct($shopCategoryId)
+    {
+        $this->shopCategoryId = $shopCategoryId;
```

Comment:
Do something like `Assertion::allIntegerish($shopCategoryId);` before assigning the value. Check `\Shopware\Bundle\SearchBundle\Condition\CategoryCondition::__construct` for reference.